packages/commonjs/src/resolve-require-sources.js

```diff
@@ -0,0 +1,169 @@
+import {
+  EXTERNAL_SUFFIX,
+  IS_WRAPPED_COMMONJS,
+  PROXY_SUFFIX,
+  wrapId,
+  WRAPPED_SUFFIX
+} from './helpers';
+import { resolveExtensions } from './resolve-id';
+
+export function getRequireResolver(extensions, detectCyclesAndConditional) {
+  const knownCjsModuleTypes = Object.create(null);
+  const requiredIds = Object.create(null);
+  const unconditionallyRequiredIds = Object.create(null);
+  const dependencies = Object.create(null);
+  const getDependencies = (id) => dependencies[id] || (dependencies[id] = new Set());
+
+  const isCyclic = (id) => {
+    const dependenciesToCheck = new Set(getDependencies(id));
+    for (const dependency of dependenciesToCheck) {
+      if (dependency === id) {
+        return true;
+      }
+      for (const childDependency of getDependencies(dependency)) {
+        dependenciesToCheck.add(childDependency);
+      }
+    }
+    return false;
+  };
+
+  // Once a module is listed here, its type (wrapped or not) is fixed any may
```

Comment:
I think there's a small typo:

```suggestion
  // Once a module is listed here, its type (wrapped or not) is fixed and may
```